### PR TITLE
BFD-3072: Migrate database from public schema to ccw

### PR DESCRIPTION
**Pull request by dondevun**
New flyway scripts have been created to migrate the contents of the public schema to the ccw schema, as well as code fixes to use the new schema.

It was originally intended for the schema name to be `bfd`, however flyway behaved erratically when that schema name was used (possibly due to the integration tests having a database named `bfd`). 

**JIRA Ticket:**
[BFD-3072](https://jira.cms.gov/browse/BFD-3072)

**User Story or Bug Summary:**

---

### What Does This PR Do?

Replace me.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] The data dictionary has been updated with any field mapping changes, if any were made.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: ops/terraform/services/base/ephemeral.tf
Comment (mjburling) [inline ops/terraform/services/base/ephemeral.tf:1]: This is an unrelated, ephemeral environment change for prod-seeded ephemeral environments.

File: ops/terraform/services/pipeline/user-data.sh.tftpl
Comment (mjburling) [inline ops/terraform/services/pipeline/user-data.sh.tftpl:1]: This is an unrelated, ephemeral environment change for prod-seeded ephemeral environments.

**Comment by aschey-forpeople** [comment]
Did a quick search through the code for any reference to `public.`. Pretty much just old migrations as expected but I did see two scripts that reference the table name with `public.` [here](https://github.com/CMSgov/beneficiary-fhir-data/blob/master/ops/ccs-ops-misc/synthetic-data/scripts/synthea-manual/generate-characteristics-file.py) and [here](https://github.com/CMSgov/beneficiary-fhir-data/blob/master/ops/ccs-ops-misc/synthetic-data/scripts/synthea-manual/pre-validate-synthea.py). Not sure if those scripts are still relevant or not.

File: ops/terraform/services/pipeline/user-data.sh.tftpl [hunk 1]
```diff
@@ -68,7 +70,9 @@ ansible-playbook \
   --extra-vars '@common_vars.json' \
   --extra-vars '@nonsensitive_pipeline_vars.json' \
   --extra-vars '@nonsensitive_shared_pipeline_vars.json' \
+%{ if pipeline_instance == "rda" }
```
Comment (mjburling) [inline ops/terraform/services/pipeline/user-data.sh.tftpl:73]: This looks like it needs another `~` to trim the whitespace.

File: ops/terraform/services/pipeline/user-data.sh.tftpl [hunk 2]
```diff
@@ -18,12 +18,14 @@ aws ssm get-parameters-by-path \
     --region us-east-1 \
     --query 'Parameters' | jq 'map({(.Name|split("/")|last): .Value})|add' > nonsensitive_pipeline_vars.json
 
+%{ if pipeline_instance == "rda" }
```
Comment (mjburling) [inline ops/terraform/services/pipeline/user-data.sh.tftpl:21]: It probably wouldn't hurt to add another whitespace trim here, too.
```suggestion
%{ if pipeline_instance == "rda" ~}
```

File: apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/SkippedRifRecord.java [hunk 3]
```diff
@@ -39,7 +39,7 @@ public class SkippedRifRecord {
       generator = "skipped_rif_records_record_id_seq")
   @SequenceGenerator(
       name = "skipped_rif_records_record_id_seq",
-      sequenceName = "skipped_rif_records_record_id_seq",
+      sequenceName = "ccw.skipped_rif_records_record_id_seq",
```
Comment (aschey-forpeople) [inline apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/SkippedRifRecord.java:42]: I found one more sequence reference here: https://github.com/CMSgov/beneficiary-fhir-data/blob/master/apps/bfd-model/bfd-model-dsl-codegen/bfd-model-dsl-codegen-plugin/src/test/resources/GenerateEntitiesFromDslMojoIT/expected/FissClaim.java#L200 does this need to be updated?

File: ops/terraform/services/pipeline/user-data.sh.tftpl (hunk 1 above)
Comment (mjburling) [inline ops/terraform/services/pipeline/user-data.sh.tftpl:73]: This looks like it needs another `~` to trim the whitespace.
```suggestion
%{ if pipeline_instance == "rda" ~}
```

File: ops/terraform/services/pipeline/user-data.sh.tftpl (hunk 2 above)
Comment (aschey-forpeople) [inline ops/terraform/services/pipeline/user-data.sh.tftpl]: I see three variables under `/bfd/prod/pipeline/sensitive/ccw` - `ccw_rif_role_external_id`, `ccw_rif_role_principal_arns`, and `s3_bucket_name`. We're sure we don't need these? They do seem to be referenced.

Comment (aschey-forpeople) [reply ops/terraform/services/pipeline/user-data.sh.tftpl]: I guess the first two are only referenced by other terraform modules but `s3_bucket_name` is references in the app configuration.

File: ops/terraform/services/base/ephemeral.tf [hunk 4]
```diff
@@ -27,28 +27,31 @@ locals {
...
-    "/bfd/${local.env}/pipeline/sensitive/rda/data_pipeline_rda_grpc_host"           = "/bfd/${local.seed_env}/pipeline/sensitive/rda/data_pipeline_rda_grpc_host"
-    "/bfd/${local.env}/pipeline/sensitive/rda/data_pipeline_rda_grpc_auth_token"     = "/bfd/${local.seed_env}/pipeline/sensitive/rda/data_pipeline_rda_grpc_auth_token"
-    "/bfd/${local.env}/pipeline/sensitive/rda/data_pipeline_rda_grpc_port"           = "/bfd/${local.seed_env}/pipeline/sensitive/rda/data_pipeline_rda_grpc_port"
-    "/bfd/${local.env}/pipeline/sensitive/rda/grpc/auth_token"                       = "/bfd/${local.seed_env}/pipeline/sensitive/rda/grpc/auth_token"
-    "/bfd/${local.env}/pipeline/sensitive/rda/grpc/port"                             = "/bfd/${local.seed_env}/pipeline/sensitive/rda/grpc/port"
-    "/bfd/${local.env}/test/pipeline/sensitive/rda/grpc/host"                        = "/bfd/${local.seed_env}/pipeline/sensitive/rda/grpc/host"
+
+    "/bfd/${local.env}/pipeline/sensitive/rda/data_pipeline_rda_grpc_host"       = local.seed_env == "prod-sbx" ? "" : "/bfd/${local.seed_env}/pipeline/sensitive/rda/data_pipeline_rda_grpc_host"
```
Comment (aschey-forpeople) [inline ops/terraform/services/base/ephemeral.tf:31]: A small comment explaining why these are empty might be nice

File: apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/SkippedRifRecord.java (hunk 3 above)
Comment (mjburling) [reply apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/SkippedRifRecord.java:42]: It's test data. Nothing to see here.

File: ops/terraform/services/base/ephemeral.tf [hunk 5]
```diff
@@ -27,28 +27,31 @@ locals {
...
     "/bfd/${local.env}/pipeline/sensitive/shared/data_pipeline_hicn_hash_pepper"     = "/bfd/${local.seed_env}/pipeline/sensitive/shared/data_pipeline_hicn_hash_pepper"
-    "/bfd/${local.env}/pipeline/sensitive/rda/data_pipeline_rda_grpc_host"           = "/bfd/${local.seed_env}/pipeline/sensitive/rda/data_pipeline_rda_grpc_host"
-    "/bfd/${local.env}/pipeline/sensitive/rda/data_pipeline_rda_grpc_auth_token"     = "/bfd/${local.seed_env}/pipeline/sensitive/rda/data_pipeline_rda_grpc_auth_token"
-    "/bfd/${local.env}/pipeline/sensitive/rda/data_pipeline_rda_grpc_port"           = "/bfd/${local.seed_env}/pipeline/sensitive/rda/data_pipeline_rda_grpc_port"
-    "/bfd/${local.env}/pipeline/sensitive/rda/grpc/auth_token"                       = "/bfd/${local.seed_env}/pipeline/sensitive/rda/grpc/auth_token"
-    "/bfd/${local.env}/pipeline/sensitive/rda/grpc/port"                             = "/bfd/${local.seed_env}/pipeline/sensitive/rda/grpc/port"
-    "/bfd/${local.env}/test/pipeline/sensitive/rda/grpc/host"                        = "/bfd/${local.seed_env}/pipeline/sensitive/rda/grpc/host"
+
```
Comment (mjburling) [inline ops/terraform/services/base/ephemeral.tf:30]:
```suggestion
# The prod-sbx environment includes an in-process server instead of a communicating with an external, gRPC host
```

**Review by mjburling — CHANGES_REQUESTED** [review]

File: ops/terraform/services/pipeline/user-data.sh.tftpl (hunk 2 above)
Comment (mjburling) [reply ops/terraform/services/pipeline/user-data.sh.tftpl]: If not in the ccw hierarchy, it's part of the shared hierarchy at this point. That will continue to be valid configuration until stories/issues subordinate to BFD-2903 are complete.

**Review by mjburling — APPROVED** [review]

**Review by meliGuzman — APPROVED** [review]
